src/App.jsx

```diff
@@ -14,45 +17,105 @@ export const goodsFromServer = [
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGoods(goods, sortField, reversedOrder) {
+  const preparedGoods = [...goods];
+
+  if (sortField || reversedOrder) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return reversedOrder ? preparedGoods.reverse() : preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversedOrder, setReversedOrder] = useState(false);
+  const goodsToDispaly
+    = getPreparedGoods(goodsFromServer, sortField, reversedOrder);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-info',
+            { 'is-light': !(sortField === SORT_FIELD_ALPHABET) },
+          )}
+          onClick={() => {
+            setSortField(SORT_FIELD_ALPHABET, reversedOrder);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-success',
+            { 'is-light': !(sortField === SORT_FIELD_LENGTH) },
+          )}
+          onClick={() => {
+            setSortField(SORT_FIELD_LENGTH, reversedOrder);
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-warning',
+            { 'is-light': !reversedOrder },
+          )}
+          onClick={() => {
+            setReversedOrder(!reversedOrder);
+          }}
+        >
+          Reverse
+        </button>
+
+        {(sortField || reversedOrder) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField('');
+              setReversedOrder(false);
+            }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here

Reply:
Got it, thanks!